src/core/persistence/src/main/java/it/geosolutions/geostore/core/dao/ldap/impl/UserDAOImpl.java

```diff
@@ -0,0 +1,260 @@
+    }
+
+    /*
+     * (non-Javadoc)
+     * 
+     * @see com.trg.dao.jpa.GenericDAOImpl#findAll()
+     */
+    @Override
+    public List<User> findAll() {
+        return ldapSearch(baseFilter, new NullDirContextProcessor());
+    }
+
+    /*
+     * (non-Javadoc)
+     * 
+     * @see com.trg.dao.jpa.GenericDAOImpl#search(com.trg.search.ISearch)
+     */
+    @SuppressWarnings("unchecked")
+    @Override
+    public List<User> search(ISearch search) {
+        if (isNested(search)) {
+            List<User> users = new ArrayList<User>();
+            for(UserGroup group :  userGroupDAO.search(getNestedSearch(search))) {
+                users.addAll(group.getUsers());
+            }
+            return users;
+        } else {
+            return ldapSearch(combineFilters(baseFilter, getLdapFilter(search, getPropertyMapper())), getProcessorForSearch(search));
+        }
+    }
+
+    /**
+     * Maps user properties to LDAP properties.
+     * @return
+     */
+    private Map<String, Object> getPropertyMapper() {
+        Map<String, Object> mapper = new HashMap<>();
+        mapper.put("name", nameAttribute);
+        for(String ldap : attributesMapper.keySet()) {
+            mapper.put(attributesMapper.get(ldap), ldap);
+        }
+        // sub-mapper for groups properties
+        if (userGroupDAO != null) {
+            mapper.put("groups", userGroupDAO.getPropertyMapper());
+        }
+        return mapper;
+    }
+
+    protected List<User> ldapSearch(String filter, DirContextProcessor processor) {
+        SearchControls controls = new SearchControls();
+        controls.setSearchScope(SearchControls.SUBTREE_SCOPE);
+        return template.search(searchBase, filter, controls, new AbstractContextMapper() {
+            int counter = 1;
+            @Override
+            protected User doMapFromContext(DirContextOperations ctx) {
+                User user = new User();
+                user.setId((long)counter++); // TODO: optionally map an attribute to the id
+                user.setEnabled(true);
+                user.setName(ctx.getStringAttribute(nameAttribute));
+                List<UserAttribute> attributes = new ArrayList<UserAttribute>();
+                for (String ldapAttr : attributesMapper.keySet()) {
+                    String value = ctx.getStringAttribute(ldapAttr);
+                    String userAttr = attributesMapper.get(ldapAttr);
+                    UserAttribute attr = new UserAttribute();
+                    attr.setName(userAttr);
+                    attr.setValue(value);
+                    attributes.add(attr);
+                }
+                user.setAttribute(attributes);
+                return user;
+            }
+            
+        }, processor);
+    }
+
+    /*
+     * (non-Javadoc)
+     * 
+     * @see com.trg.dao.jpa.GenericDAOImpl#merge(java.lang.Object)
+     */
+    @Override
+    public User merge(User entity) {
+        // DO NOTHING: PERSITENCE IS NOT SUPPORTED
```

Comment:
Theoretically these write operations should fail instead of faking success (rising an exception like "unsupported operation") ? I know this is a safest solution in terms of compatibility, at the expense of the consistency of the operations, Did you already tried?

Reply:
I put execeptions initially, but I had to remove them because persistence was called in some 
weird cases, I will check if those are not an issue anymore.

Reply:
Looks like it works also throwing exception, so I did another commit for this